extensions/tinyfaces-nft/src/tinyfaces.tsx

```diff
@@ -0,0 +1,229 @@
+  MOUTH: 'Eg: "Bubblegum", "Face Mask", "Mustache"',
+  OWNER: 'Eg: "ped.eth", "0x1d1A1...cB751"',
+};
+
+export default function Command() {
+  return (
+    <WagmiConfig client={client}>
+      <Tiny />
+    </WagmiConfig>
+  );
+}
+
+export function Tiny() {
+  const [searchText, setSearchText] = useState<undefined | string>(undefined);
+  const [searchType, setSearchFilter] = useState<SearchFilters>("TOKEN_ID");
+  const [ensAddress, setEnsAddress] = useState<undefined | string>(undefined);
+  const [lookupENSAddress, setLookupENSAddress] = useState<boolean>(false);
+
+  const { isLoading: isLoadingEnsAddress } = useEnsAddress({
+    name: searchText,
+    enabled: lookupENSAddress,
+    onSuccess: (address) => address && setEnsAddress(address),
+  });
+
+  const isSearchingByQueryString = searchType !== "TOKEN_ID" && searchType !== "OWNER";
+
+  const { isLoading: isLoadingTokenId = false, data: tokenIdData } = useFetch<TokenData>(
+    `${API_ENDPOINT}/token/${searchText}`,
+    {
+      execute: searchType === "TOKEN_ID" && Boolean(searchText),
+    }
+  );
+  const { isLoading: isLoadingName, data: nameData } = useFetch<PaginatedData>(
+    `${API_ENDPOINT}/token?page=0&limit=100&${
+      isSearchingByQueryString && SEARCH_QUERY_STRINGS[searchType]
+    }=${searchText}`,
+    {
+      execute: isSearchingByQueryString && Boolean(searchText),
+    }
+  );
+  const { isLoading: isLoadingOwner, data: ownerData } = useFetch<OwnerData>(`${API_ENDPOINT}/address/${ensAddress}`, {
+    execute: searchType === "OWNER" && Boolean(ensAddress),
+  });
+
+  useEffect(() => {
+    if (searchType === "OWNER" && searchText) {
+      if (searchText?.includes(".eth")) setLookupENSAddress(true);
+      if (isValidEthAddress(searchText)) setEnsAddress(searchText);
+    }
+  }, [searchText, searchType]);
+
+  let isLoading;
+
+  if (searchType === "TOKEN_ID") {
+    isLoading = isLoadingTokenId && Boolean(searchText);
+  }
+  if (searchType === "NAME") {
+    isLoading = isLoadingName && Boolean(searchText);
+  }
+  if (searchType === "OWNER") {
+    isLoading = isLoadingEnsAddress || (isLoadingOwner && Boolean(ensAddress));
+  }
+
+  return (
+    <List
+      throttle
+      isShowingDetail
+      isLoading={isLoading}
+      searchBarPlaceholder={SEARCH_PLACEHOLDERS[searchType]}
+      searchText={searchText}
+      onSearchTextChange={setSearchText}
+      searchBarAccessory={
+        <SearchTypeDropdown
+          searchTypes={Object.keys(SEARCH_TYPES) as SearchQueryStrings[]}
+          onChange={(searchFilter) => {
+            setSearchText("");
+            setSearchFilter(searchFilter);
+          }}
+        />
+      }
+    >
+      {tokenIdData && Boolean(searchText) && (
+        <List.Item
+          title={tokenIdData.name}
+          accessories={[{ text: `#${tokenIdData.token_id}` }]}
+          icon={{ source: `${tokenIdData.direct_url}?width=120&height=120`, mask: Image.Mask.Circle }}
```

Comment:
The last thing I can think of if to have a [local fallback image](https://developers.raycast.com/api-reference/user-interface/icons-and-images#image.fallback) if images somehow don't load - could you add that and I think we're ready 🙂

I'm looking forward to testing this extension again 🔥

Reply:
sounds good, done!